src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -160,30 +422,87 @@ private void AddToGroup(object obj)
             }
         }
 
-        [JsonIgnore]
-        public Double FontSize
+        private bool CanAddGroupToGroup(object obj)
         {
-            get
+            // First make sure this group is selected
+            // and that it does not already belong to
+            // another group
+            if (!this.AnnotationModel.IsSelected ||
+                this.AnnotationModel.BelongsToGroup) 
             {
-                return annotationModel.FontSize;
+                return false;
             }
-            set
+
+            var selectedAnnotationModels = DynamoSelection.Instance.Selection
+                    .OfType<AnnotationModel>()
+                    .Where(model => model.GUID != this.AnnotationModel.GUID);
+
+            // Then we make sure that there are any other
+            // AnnotationModels selected
+            if (!selectedAnnotationModels.Any()) return false;
+
+            // Lastly we make sure that non of the selected
+            // groups (except this one) already has nested groups
+            // and there are at least one of the groups that does
+            // not already belong to another group.
+            return !selectedAnnotationModels.Any(x => x.HasNestedGroups) &&
+                !selectedAnnotationModels.All(x => x.BelongsToGroup);
+                    
+        }
+
+        private void AddGroupToGroup(object obj)
+        {
+            if (annotationModel.IsSelected)
             {
-                annotationModel.FontSize = value;                
+                var selectedModels = DynamoSelection.Instance.Selection
+                    .OfType<AnnotationModel>()
+                    .Where(x => x.GUID != this.AnnotationModel.GUID && !x.BelongsToGroup);
+
+                foreach (var model in selectedModels)
+                {
+                    WorkspaceViewModel.DynamoViewModel.AddGroupToGroupModelCommand.Execute(this.AnnotationModel.GUID);
```

Comment:
I saw you have a constructor to take a collection, I guess not using here because we have follow-ups below to do in addition?